app/controllers/internal_api/v1/pillars_controller.rb

```diff
@@ -22,9 +22,9 @@ def pillar_contents
   def registry_contents
     registries = DockerRegistry.all.map do |reg|
```

Comment:
You need to fetch all the registries with `mirror: nil`.

Reply:
i think here we need all registries. also the ones without a mirror, as this would be also a valid option on the `daemon.json` too

Reply:
Nope. IIRC you cannot have the same URL being used as `Prefix` and `URL` inside `daemon.json`. @vrothberg could confirm that.